kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -359,34 +369,43 @@ SetOfKeyValuePairs DBAdapter::lastReachableBlockDbUpdates(const SetOfKeyValuePai
   }
 
   // Split the keys: NonProvable and Provable Keys
-  SetOfKeyValuePairs non_provable_kvs;
-  SetOfKeyValuePairs provable_kvs = updates;
+  SetOfKeyValuePairs nonProvableKvPairs;
+  SetOfKeyValuePairs provableKvPairs = updates;
   for (const auto &k : nonProvableKeySet_) {
     if (deletes.find(k) != std::cend(deletes)) {
       throw std::runtime_error{"The deletes key set contains a non-provable key: " + k.toString()};
     }
-    auto iter = provable_kvs.find(k);
-    if (iter != std::cend(provable_kvs)) {
-      non_provable_kvs.emplace(iter->first, iter->second);
-      provable_kvs.erase(iter);
+    auto iter = provableKvPairs.find(k);
+    if (iter != std::cend(provableKvPairs)) {
+      nonProvableKvPairs.emplace(iter->first, iter->second);
+      provableKvPairs.erase(iter);
     }
   }
 
   // If there are no updates and no actual deletes, do not update the tree and do create an empty block.
-  if (!provable_kvs.empty() || !actuallyDeleted.empty()) {
+  if (!provableKvPairs.empty() || !actuallyDeleted.empty()) {
     // Key updates.
     const auto updateBatch =
-        smTree_.update(provable_kvs, KeysVector{std::cbegin(actuallyDeleted), std::cend(actuallyDeleted)});
+        smTree_.update(provableKvPairs, KeysVector{std::cbegin(actuallyDeleted), std::cend(actuallyDeleted)});
     const auto batchDbUpdates = batchToDbUpdates(updateBatch, blockId);
     dbUpdates.insert(std::cbegin(batchDbUpdates), std::cend(batchDbUpdates));
   }
 
   // Block node with updates and actually deleted keys.
   dbUpdates[DBKeyManipulator::genBlockDbKey(blockId)] =
-      createBlockNode(provable_kvs, actuallyDeleted, blockId, parentBlockDigestFuture.get());
-
-  for (const auto &[k, v] : non_provable_kvs) {
-    dbUpdates[DBKeyManipulator::genNonProvableBlockDbKey(blockId, k)] = v;
+      createBlockNode(provableKvPairs, actuallyDeleted, blockId, parentBlockDigestFuture.get());
+
+  // Marking non-provable keys as stale
```

Comment:
Nit: I think that comment belongs a bit lower as here we are both adding the non-provable key and marking it stale.